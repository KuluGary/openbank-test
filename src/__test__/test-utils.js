import React from "react"
import { render as rtlRender } from "@testing-library/react"
import { Provider } from "react-redux"

import reduxStore from "store";

/**
 * Wraps the component in a mock of the redux store provider
 * @param {Component} ui - the UI component passed on render
 * @param {Object} renderOptions - the render options passed 
 */
const render = (ui, { initialState, store = reduxStore, ...renderOptions } = {}) => {
    const Wrapper = ({ children }) => {
        return <Provider store={store}>{children}</Provider>
    }

    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react"; // export full testing-library components
export { render }; // export provider wrapped render
