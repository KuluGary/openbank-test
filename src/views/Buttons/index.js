import React from 'react'
import PropTypes from 'prop-types'
import { useSelector } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { Button } from '@material-ui/core'
import NavigateNextIcon from '@material-ui/icons/NavigateNext'

/**
 * Component that renders buttons and handles navigation through
 * the wizard
 * @param {Number} activeStep - current index of active step
 * @param {Function} handleBack - handler function for backward navigation
 * @param {Function} handleNext - handler function for forward navigation
 */
export default function Buttons ({ activeStep, handleBack, handleNext }) {
  const { t } = useTranslation()
  const { terms } = useSelector(state => state.terms)
  const { password, passwordError, passwordHintError } = useSelector(state => state.password)

  const step1Validator = () => {
    return !!terms
  }

  const step2Validator = () => {
    return !!(password.length <= 0 || passwordError || passwordHintError)
  }

  switch (activeStep) {
    case 0:
      return (
        <>
            <Button data-testid="left-button" onClick={() => console.log('Cancela el wizard')}>
                {t('wizard.buttons.cancel')}
            </Button>
            <Button
                data-testid="right-button"
                variant="contained"
                color="secondary"
                disabled={!step1Validator()}
                onClick={handleNext}
            >
                <>
                    {t('wizard.buttons.next')}
                    <NavigateNextIcon />
                </>
            </Button>
        </>
      )
    case 1:
      return (
        <>
            <Button onClick={handleBack} data-testid="left-button">
                {t('wizard.buttons.back')}
            </Button>
            <Button
                variant="contained"
                color="secondary"
                disabled={step2Validator()}
                onClick={handleNext}
                data-testid="right-button"
            >
                <>
                    {t('wizard.buttons.next')}
                    <NavigateNextIcon />
                </>
            </Button>
        </>
      )
    default:
      return (
        <Button onClick={handleBack} data-testid="right-button">
            {t('wizard.buttons.back')}
        </Button>
      )
  }
}

Buttons.propTypes = {
  activeStep: PropTypes.number,
  handleBack: PropTypes.func,
  handleNext: PropTypes.func
}
