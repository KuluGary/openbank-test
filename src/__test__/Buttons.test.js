import React from 'react';
import Buttons from 'views/Buttons';

import { render, cleanup, fireEvent } from './test-utils';
import '@testing-library/jest-dom/extend-expect';

const props = {
  activeStep: 0,
  handleBack: () => {},
  handleNext: () => {}
}

describe('Buttons', () => {
  afterEach(cleanup);

  it('matches snapshot', () => {
    const div = document.createElement('div');
    const component = render(<Buttons {...props} />, div);
    expect(component.container).toMatchSnapshot();
  })

  it('is disabled if user has not accepted privacy terms', () => {
    const { getByTestId } = render(<Buttons {...props} />)
    const rightButton = getByTestId('right-button')

    expect(rightButton.disabled).toEqual(true);
  })
})

