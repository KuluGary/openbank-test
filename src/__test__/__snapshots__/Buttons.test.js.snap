// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons matches snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    data-testid="left-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      wizard.buttons.cancel
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary Mui-disabled Mui-disabled"
    data-testid="right-button"
    disabled=""
    tabindex="-1"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      wizard.buttons.next
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </span>
  </button>
</div>
`;
